eng/resolveContract.targets

```diff
@@ -62,6 +63,22 @@
     <_ApiCompatLibReplacementString Condition="'$(MSBuildProjectName)' == 'System.Private.CoreLib'">lib/$(NetCoreAppCurrent)/$2</_ApiCompatLibReplacementString>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <IsRuntimeAndReferenceAssembly Condition="'$(IsRuntimeAndReferenceAssembly)' == '' and
+                                              '$(HasMatchingContract)' != 'true' and
+                                              '$(IsPrivateAssembly)' != 'true'">true</IsRuntimeAndReferenceAssembly>
+  </PropertyGroup>
+
+  <!-- Disable suppression validation for packable projects that have a corresponding reference source project.
+       That is necessary as APICompat is invoked twice, once for the ref <-> src comparision and then again
+       for the package validation (which doesn't include reference assemblies). As both operations don't have
+       all the inputs available, some suppressions might only apply to one or the other and hence unnecessary
+       suppressions can't be determined. -->
+  <PropertyGroup Condition="'$(IsPackable)' == 'true' and '$(IsRuntimeAndReferenceAssembly)' != 'true'">
+    <ApiCompatPreserveUnnecessarySuppressions>true</ApiCompatPreserveUnnecessarySuppressions>
+    <ApiCompatPermitUnnecessarySuppressions>true</ApiCompatPermitUnnecessarySuppressions>
```

Comment:
These two properties sound very similar. Why two?

Reply:
APICompat is used in dotnet/runtime but also in various other repositories. these two new settings are on by default but need to be disabled at least in runtime for a few projects (mentioned that in the code comment in resolveContract.targets). The feature flags are distinct as the latter allows unnecessary suppressions to exist in the suppression files and the former trims out unnecessary suppressions when re-generating a suppression file via the `/p:GenerateAPICompatSuppressionFile=true` switch.